### DISCUSSION TITLE
Naming of foreign key columns

### DISCUSSION THREAD

**Discussion by davidolrik** [piccolo-orm/piccolo#206] [Q&A]
Given the default schema example:

```python
from piccolo.table import Table
from piccolo.columns import ForeignKey, Integer, Varchar


class Manager(Table):
    name = Varchar(length=100)


class Band(Table):
    name = Varchar(length=100)
    manager = ForeignKey(references=Manager)
    popularity = Integer()
```

This will create a `band` table with a `manager` column and a `manager` property.

Is it possible to get a `manager_id` in the database while still getting the "pretty" property name?

If not, maybe something like this could be implemented:

```python
class Band(Table):
    name = Varchar(length=100)
    manager = ForeignKey(column="manager_id", references=Manager)
    popularity = Integer()
```

**Comment by dantownsend** [comment]
There's no API at the moment for mapping from Python attribute names to SQL column names. The names have to be the same.

I like the idea you proposed:

```python
class Band(Table):
    manager = ForeignKey(column_name="manager_id", references=Manager)
```

It could be used for any column type too.

It could be the column equivalent of specifying a custom tablename:

```python
class Band(Table, tablename="my_table"):
    ...
```

I don't mind converting this into an issue, so it can be implemented at some point.

**Reply by davidolrik** [reply]
Good idea to make it a general thing! - That has the benefit of nice code on top of badly named legacy databases 👍🏽
